packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -933,6 +933,7 @@ class _DraggableScrollableSheetScrollPosition extends ScrollPositionWithSingleCo
     // extent.activePositionCount as if the activity were ours.
     if (other.activity!.isScrolling) {
       ++extent.activePositionCount;
+      _isActive = true;
```

Comment:
At this point I think I've introduced enough bugs in "active" state management that it might be worth coupling `_DraggableScrollableSheetScrollPosition` to `_DraggableScrollableSheetScrollController` after all, and re-evaluating the status of the scroll positions on the fly. (Related: [Effective Dart](https://dart.dev/guides/language/effective-dart/usage#avoid-storing-what-you-can-calculate).) The `getExtent` `final` field is already a manifestation of this coupling.